tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs

```diff
@@ -123,24 +126,32 @@ public async Task StartFileActivationFlowAsync(StorageFile file)
         await _mainPageViewModel.StartConfigurationFileAsync(file);
     }
 
-    public void StartCreationFlowAsync()
+    public void StartCreationFlowAsync(string originPage)
     {
         Orchestrator.FlowPages = [_mainPageViewModel];
+
+        // this method is only called when the user clicks a button that redirects them to 'Create Environment' flow in the setup flow.
+        TelemetryFactory.Get<ITelemetry>().Log(
+            "Create_Environment_button_Clicked",
+            LogLevel.Critical,
+            new EnvironmentRedirectionUserEvent(navigationAction: _creationFlowNavigationParameter, originPage),
+            relatedActivityId: Orchestrator.ActivityId);
+
         _mainPageViewModel.StartCreateEnvironment(string.Empty);
     }
 
     public void OnNavigatedTo(NavigationEventArgs args)
     {
         // The setup flow isn't setup to support using the navigation service to navigate to specific
         // pages. Instead we need to navigate to the main page and then start the creation flow template manually.
-        var parameter = args.Parameter?.ToString();
+        var parameters = $"{args.Parameter}".Split(';');

```

Comment:
Adding this functionality drastically modifies how DevHome interacts with `OnNavigatedTo`  Every place in DevHome now needs to know that `OnNavigatedTo` will try to split `args.Parameter` on ;'s.

What is the purpose of doing this?

Reply:
I'm not sure what you mean here. This doesn't modify how `OnNavigatedTo` works in other places in Dev Home. This is only in the Setup flow, and is used for us to allow a page from anywhere in Dev Home to navigate to the setup flow, start a specific flow and provide the page that the navigation came from so we can add it to the telemetry. E.g the user clicks the create environment button in the environments page, which navigated to the setupflow, who then checks the parameters.  This change only affects the setup flow, and the only thing in Dev Home today that performs/uses this functionality is environments page. But if I'm missing something please let me know

Reply:
spoke to Darren offline and agreed to add more information on why we're using a semicolon delimiter for the string and moved it to be inside the `if` statement. This is so future devs don't need to always be tied to using a semicolon delimited string for things other than environment creation in the  `OnNavigatedTo` method